pkg/services/auth/jwt/verification.go

```diff
@@ -1,60 +1,80 @@
 package jwt
 
 import (
-	"encoding/json"
+	"context"
+	"encoding/base64"
+	"errors"
 	"fmt"
 	"reflect"
+	"strings"
 	"time"
 
 	"gopkg.in/square/go-jose.v2/jwt"
 
+	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/setting"
 )
 
-func (s *AuthService) initClaimExpectations() error {
-	if err := json.Unmarshal([]byte(s.Cfg.JWTAuthExpectClaims), &s.expect); err != nil {
-		return err
+func ProvideVerificationService(cfg *setting.Cfg) (*VerificationService, error) {
```

Comment:
Seems to me `cfg` is never used within this service so please remove it. 

Nit. This service might be converted into just a utility function `Verify` since it receives everything it needs via arguments (besides the logger). I can understand having a struct/interface for this can be beneficial from a mocking standpoint, but mocking just a function in Go is normally easier.